dvc/objects/fs/path.py

```diff
@@ -80,7 +80,7 @@ def parents(self, path):
         )
 
     def name(self, path):
-        return self.parts(path)[-1]
+        return path.rsplit(self.flavour.sep, maxsplit=1)[-1]
```

Comment:
This doesn't handle `foo/` unlike the old implementation 🙁

Reply:
Updated the implementation, also added some tests, currently waiting for benchmarks